src/rust/src/contract.rs

```diff
@@ -82,6 +82,17 @@ fn xdr_to_rust_buf<T: WriteXdr>(t: &T) -> Result<RustBuf, HostError> {
     Ok(RustBuf { data })
 }
 
+fn event_to_diagnostic_event_to_rust_buff(
```

Comment:
nit: to_rust_buf? (for consistency with other names). Also `event_to_diagnostic_event_rust_buf` is a bit more readable IMO

Reply:
Updated to `event_to_diagnostic_event_rust_buf `